algebird-core/src/main/scala/com/twitter/algebird/QTree.scala

```diff
@@ -38,23 +38,21 @@ package com.twitter.algebird
  */
 
 object QTree {
-  def apply[A: Monoid](kv: (Double, A), level: Int = -16): QTree[A] = {
+  def apply[A](kv: (Double, A), level: Int = -16): QTree[A] =
     QTree(math.floor(kv._1 / math.pow(2.0, level)).toLong,
       level,
       1,
       kv._2,
       None,
       None)
-  }
 
-  def apply[A: Monoid](kv: (Long, A)): QTree[A] = {
+  def apply[A](kv: (Long, A)): QTree[A] =
     QTree(kv._1,
       0,
       1,
       kv._2,
       None,
       None)
-  }
 
   /**
    * The common case of wanting a count and sum for the same value
```

Comment:
do you think it is worth using:
```scala
   /**
    * If you are sure you only care about the approximate histogram
    * features of QTree, you can save some space by using QTree[Unit]
    */
   def offset(o: Long): QTree[Unit] = apply(o -> ())
```
At least Unit is zero bytes serialized, and points to the same memory. We use this internally in Tsar which is only concerned with the counts.